pkg/svcat/svcat.go

```diff
@@ -49,7 +55,13 @@ func NewApp(kubeConfig, kubeContext string) (*App, error) {
 
 // configForContext creates a Kubernetes REST client configuration for a given kubeconfig context.
 func configForContext(kubeConfig, kubeContext string) (*rest.Config, error) {
-	config, err := kube.GetConfig(kubeContext, kubeConfig).ClientConfig()
+	var config *rest.Config
```

Comment:
I suggest changing the signature of `svcat.NewApp` to accept a service catalog client interface. Then cmd/svcat can keep it's plugin logic where it is and this the conditional plugin mode handling would stay in that package (we already have similar if/else logic in main today):

https://github.com/kubernetes-incubator/service-catalog/blob/e15b73719911853d3755b71f3d8b26b21296d0a3/cmd/svcat/main.go#L73

Maybe something like

```go
config, err := loadConfig(opts.KubeConfig, opts.KubeContext)
app, err := svcat.NewApp(config)

// ...

func loadConfig(kubeConfig, kubeContext string) (clientset.Interface, error) {
  if plugin.IsPlugin() {
    return pluginutils.InitConfig()
  } else {
    return svcat.GetKubeConlient(kubeconfig, kubecontext)
  }
}
```

```

Reply:
That sounds like  a much better idea than my jank. Redid it, tell me what you think.